src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -228,10 +229,15 @@ public override ComputedStaticFieldLayout ComputeStaticFieldLayout(DefType defTy
             return result;
         }
 
-        private ref StaticsBlock GetStaticsBlockForField(ref ComputedStaticFieldLayout layout, FieldDesc field)
+        protected virtual ref StaticsBlock GetStaticsBlockForField(ref ComputedStaticFieldLayout layout, FieldDesc field)
```

Comment:
This extension point seems to be similar to the [extension point](https://github.com/dotnet/corert/pull/1501) @davidwrighton added for computing whether a field has a GC static base.

Maybe we can get away with just that and we don't need this to be virtual? (i.e. override what `ComputeHasGCStaticBase` says for thread static fields.)